src/components/modal/modal.tsx

```diff
@@ -0,0 +1,43 @@
+import { useRef, type FC, type ReactNode } from "react";
+import { createPortal } from "react-dom";
+import styles from "./modal.module.css";
+import { HiMiniXMark } from "react-icons/hi2";
+import cx from "classnames";
+
+interface Props {
+	children?: ReactNode;
+	open?: boolean;
+	onClose?: () => void;
+}
+
+export const Modal: FC<Props> = ({ onClose, open, children }) => {
+	const ref = useRef<HTMLDialogElement>(null);
+
+	const handleOutsideClick = () => {
+		onClose?.();
+	};
+	const handleDialogClick = (e: React.MouseEvent) => {
+		e.stopPropagation();
+	};
+
+	if (!open) return null;
+	return createPortal(
+		<div className={cx(styles.overlay)} onClick={handleOutsideClick}>

```

Comment:
```suggestion
		<div className={styles.overlay} onClick={onClose}>
```
если один класс - не надо cx. и onClose можно сразу прокинуть, если кроме него ниче не делается по клику